src/aiortc/contrib/media.py

```diff
@@ -382,3 +383,160 @@ async def __run_track(self, track, context):
                 return
             for packet in context.stream.encode(frame):
                 self.__container.mux(packet)
+
+
+class MultiplexedMediaSource:
+    def __init__(self, track):
+        """
+        Create a thread which runs continu to get frames from the camera
+        Frames itself are dispatched to multiple queues
+        Clients can add itself and remove itself to get new frames
+        """
+        self.__track = track
+
+        self.__streams = set()
+        self.__stream_lock = threading.Lock()
+
+        self.__thread_quit = threading.Event()
+        self.__thread: threading.Thread = threading.Thread(
+            name="Multiplexed source",
+            target=self.worker,
+            args=(
+                asyncio.get_event_loop(),
+                self.__track,
+                self.__streams,
+                self.__stream_lock,
+                self.__thread_quit
+            )
+        )
+        self.__thread.start()
+
+    @property
+    def kind(self):
+        return self.__track.kind
+
+    async def stop(self):
+        """
+        Stop camera thread and release camera resource
+        :return:
+        """
+        self.__thread_quit.set()
+        loop = asyncio.get_event_loop()
+        await loop.run_in_executor(None, self.__thread.join)
+        self.__track.stop()
+
+    def add_peer_connection(self, peer_connection):
```

Comment:
This looks like it's the wrong way round. We want an API which returns the `MultiplexedMediaStreamTrack`, regardless of what we use it for.

One possible usecase: in the `server` example it would be nice to be able to take the received video and both send it to a `MediaRecorder` and send it through the "transform track" to echo it back to the browser.